improver_tests/blending/spatial_weights/test_SpatiallyVaryingWeightsFromMask.py

```diff
@@ -749,41 +320,35 @@ def test_no_fuzziness_with_one_dimensional_weights(self):
     def test_fuzziness_no_one_dimensional_weights(self):
         """Test a simple case where we have some fuzziness in the spatial
         weights and no adjustment from the one_dimensional weights."""
-        plugin = SpatiallyVaryingWeightsFromMask(fuzzy_length=2)
         self.one_dimensional_weights_cube.data = np.ones((3))
         expected_result = np.array(
             [
-                [[0.5, 0.0, 0.5], [0.707107, 0.5, 0.707107]],
-                [[0.0, 0.0, 0.5], [0.0, 0.5, 0.707107]],
-                [[1.0, 1.0, 1.0], [1.0, 1.0, 1.0]],
+                [[0.25, 0.0, 0.166667], [0.353553, 0.166667, 0.235702]],
+                [[0.00, 0.0, 0.166667], [0.000000, 0.166667, 0.235702]],
+                [[0.75, 1.0, 0.666667], [0.646447, 0.666667, 0.528595]],
             ],
             dtype=np.float32,
         )
-        result = plugin.process(
-            self.cube_to_collapse,
-            self.one_dimensional_weights_cube,
-            "forecast_reference_time",
+        result = self.plugin.process(
+            self.cube_to_collapse, self.one_dimensional_weights_cube,
         )
         self.assertArrayAlmostEqual(result.data, expected_result)
         self.assertEqual(result.metadata, self.cube_to_collapse.metadata)
 
     @ManageWarnings(ignored_messages=["Collapsing a non-contiguous coordinate."])
     def test_fuzziness_with_one_dimensional_weights(self):
```

Comment:
This test made me think a bit because of the different but overlapping masks in different slices of the array. Do you think we need a test to show what happens if one slice has masked data in rather than two? The second renormalisation should redistribute weight to both of the other slices evenly for the points that have been scaled with the fuzzy weights.

Reply:
Another edge case to consider. What happens if you only had two slices, and both had missing data in gridboxes either side of a point that was valid in both slices. Then this middle point would be reduced in both slices and the renormalisation wouldn't change it (I think), so you'd end up with weights that didn't add up to 1. I think this would be ok because the weighted blending code where the weights are used would renormalise and reweight them again?
e.g. 
[[--, 0.5, 0.5], [0.5, 0.5, --]] might result in weights [[0, 0.25, 1][0, 0.25, 1]]

Reply:
The second edge case, as discussed, I think is dealt with fine due to the initial normalisation.  For the case of 2 unmasked and 1 masked slice, I've now added a unit test to cover this.